centreon/src/Core/Dashboard/Application/UseCase/PartialUpdateDashboard/PartialUpdateDashboardRequest.php

```diff
@@ -24,22 +24,222 @@
 namespace Core\Dashboard\Application\UseCase\PartialUpdateDashboard;
 
 use Core\Common\Application\Type\NoValue;
+use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\Request\PanelLayoutRequestDto;
 use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\Request\PanelRequestDto;
 use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\Request\RefreshRequestDto;
+use Core\Dashboard\Application\UseCase\PartialUpdateDashboard\Request\ThumbnailRequestDto;
+use Core\Dashboard\Infrastructure\Model\RefreshTypeConverter;
+use Symfony\Component\Validator\Constraints as Assert;
 
 final class PartialUpdateDashboardRequest
 {
     /**
-     * @param NoValue|string $name
-     * @param NoValue|string $description
-     * @param NoValue|array<PanelRequestDto> $panels
-     * @param NoValue|RefreshRequestDto $refresh
+     * @param string|null $name
+     * @param string|null $description
+     * @param array{type: string, interval: int} $refresh
+     * @param array{id?: int, name: string, directory: string} $thumbnail
+     * @param array<array{
+     *    id?: ?int,
+     *    name: string,
+     *    layout: array{
+     *        x: int,
+     *        y: int,
+     *        width: int,
+     *        height: int,
+     *        min_width: int,
+     *        min_height: int
+     *    },
+     *    widget_type: string,
+     *    widget_settings: string,
+     *}> $panels
      */
     public function __construct(
-        public NoValue|string $name = new NoValue(),
-        public NoValue|string $description = new NoValue(),
-        public NoValue|array $panels = new NoValue(),
-        public NoValue|RefreshRequestDto $refresh = new NoValue(),
+        #[Assert\Type('string')]
+        #[Assert\Length(min: 1, max: 200)]
+        public mixed $name = null,
```

Comment:
mixed all the time for all the properties of this class ??

Reply:
Class properties cant be hard typed in that case since payload is multipart/form-data every data sent is a string. Furthermore if type is hard typed errors returned for validation will be the consequence of the incapacity of symfony to instanciate the request dto through the maprequest payload process. Mixed here forces the validation through the symfony validator (annotations)

Reply:
In my opinion, this is not the right way. We're modifying the principles of clean architecture to integrate Symfony components. Instead, we should take the time to study a better method for implementing this component. The same goes for events. I think we're going too fast, and the architecture is likely to be messy.
I don't think it's good practice to use a feature to integrate components without studying them in advance. We're going too fast. What's more, we end up with a PartialUpdateDashboardRequestDto class with some strange types...
In short, I don't think it's a good idea to integrate components in this way. We should take the time to study them beforehand and implement them properly.
But if the final decision is to do it this way, then I'll do an approve without really agreeing.

Reply:
We have found a solution, thanks for your change ;)